Cargo.toml

```diff
@@ -66,3 +66,6 @@ path="examples/superstreams/send_super_stream.rs"
 name="environment_deserialization"
 path="examples/environment_deserialization.rs"
 
+[[bin]]
+name = "perf-producer"
+path = "src/bin/perf-producer.rs"
```

Comment:
As this file is doing perf-tests any possibility to move it in the tests folder? (maybe creating a folder like performances)

Reply:
We have different possibilities:
- put it under `bench` folder where, typically, you put performance tests, commonly developed with `criterion` crate. Anyway, this actually isn't a benchmark in the same sense of `criterion` which invokes the function multiple time and measure the duration.
- put it under `tests` folder and threat it as a integration test. Anyway, as a test, the output is not shown to the developer without the appropriate flag (`--nocapture`).
- put it under `bin` and threat is as a binary, as in this PR. You can run it using `cargo run --release --bin perf-producer` and print the output easily.

Anyway, I don't have a strong opinion on that and I can follow a different solution. let me know how you want to proceed.